### ISSUE TITLE
Php serialize/__sleep/unserialize exchange error

### ISSUE THREAD

**Issue by brmn** [ccxt/ccxt#1080]
- OS: linux
- Programming Language: php
- CCXT version: 1.10.562
- Exchange:
- Method:

```
        $exchange = new \ccxt\poloniex();
        $exchange->publicGetReturnTicker();//ok
        $serialized = serialize($exchange);
        $unserializedExchange = unserialize($serialized);
        $unserializedExchange->publicGetReturnTicker();//publicGetReturnTicker not found
```
 it is because of Exchange::__sleep() that dont serialize dynamically created methods and resources

suggest 1: implement __wakeup method like this
```
public function __wakeup()
    {
        $this->curl        = curl_init ();
        if ($this->api)
            $this->define_rest_api ($this->api, 'request');
    }
```

suggest 2: move curl_init from __construct() to fetch()

**Comment by kroitor** [comment]
Related to #930

**Comment by kroitor** [comment]
Fixed in #1093. Thanks to @kornrunner !
